### PR TITLE
feat: allow to user a termenv renderer on bubbletea middleware

### PR DESCRIPTION
**Pull request by caarlos0**
this allows the bubbletea middleware to query the user terminal instead of using the server (or the default).

It'll also give the user a `*lipgloss.Renderer` which can be used with lipgloss directly, etc.

Tried to make it so it isn't a breaking change.

**TODO**: 

- [ ] write tests

Example is very based on @aymanbagabas & @meowgorithm work on https://github.com/charmbracelet/lipgloss/blob/master/examples/ssh/main.go

closes #71 #45
needs ~~https://github.com/charmbracelet/bubbletea/pull/720~~ https://github.com/charmbracelet/bubbletea/pull/715

### CODE REVIEW

File: bubbletea/tea.go [hunk 1]
```diff
@@ -67,32 +120,90 @@ func MiddlewareWithProgramHandler(bth ProgramHandler, cp termenv.Profile) wish.M
...
+	o := &sshOutput{
+		Session: sess,
+		tty:     tty,
 	}
+	environ := sess.Environ()
+	environ = append(environ, fmt.Sprintf("TERM=%s", sshPty.Term))
+	e := &sshEnviron{environ: environ}
+	// We need to use unsafe mode here because the ssh session is not running
+	// locally and we already know that the session is a TTY.
+	return termenv.NewOutput(o, termenv.WithUnsafe(), termenv.WithEnvironment(e), termenv.WithColorCache(true))
```
Comment (aymanbagabas) [inline bubbletea/tea.go:208]: This is basically a hack and would be as simple as passing the session to termenv with this https://github.com/muesli/termenv/pull/122

Comment (caarlos0) [reply bubbletea/tea.go:208]: oh yes that would make it a lot better!

**Comment by caarlos0** [comment]
will wait https://github.com/muesli/termenv/pull/122 before working more on this one

**Comment by caarlos0** [comment]
depends on https://github.com/charmbracelet/bubbletea/pull/739

**Review by skydiver1980 — APPROVED** [review]

**Comment by caarlos0** [comment]
superseded by #197

**Pull request closed without merging**